Remove JavaScriptCore/bindings

Constellation opened the pull request:
#### 6b0de0073377b856dda88c4bbf010294b36a5f58
<pre>
Remove JavaScriptCore/bindings
<a href="https://bugs.webkit.org/show_bug.cgi?id=256033">https://bugs.webkit.org/show_bug.cgi?id=256033</a>
rdar://problem/108596608

Reviewed by Darin Adler.

All files under JavaScriptCore/bindings were Deprecated::.

1. Remove ScriptValue and ScriptObject. Use Strong&lt;&gt; directly. We do not need these wrappers anymore.
2. Move ScriptFunctionCall to inspector/ since it is only used for inspector code, and still it is useful.

* Source/JavaScriptCore/CMakeLists.txt:
* Source/JavaScriptCore/JavaScriptCore.xcodeproj/project.pbxproj:
* Source/JavaScriptCore/Sources.txt:
* Source/JavaScriptCore/bindings/ScriptObject.cpp: Removed.
* Source/JavaScriptCore/bindings/ScriptObject.h: Removed.
* Source/JavaScriptCore/bindings/ScriptValue.cpp: Removed.
* Source/JavaScriptCore/bindings/ScriptValue.h: Removed.
* Source/JavaScriptCore/inspector/InjectedScript.cpp:
(Inspector::InjectedScript::InjectedScript):
(Inspector::InjectedScript::execute):
(Inspector::InjectedScript::evaluate):
(Inspector::InjectedScript::awaitPromise):
(Inspector::InjectedScript::callFunctionOn):
(Inspector::InjectedScript::evaluateOnCallFrame):
(Inspector::InjectedScript::getFunctionDetails):
(Inspector::InjectedScript::functionDetails):
(Inspector::InjectedScript::getPreview):
(Inspector::InjectedScript::getProperties):
(Inspector::InjectedScript::getDisplayableProperties):
(Inspector::InjectedScript::getInternalProperties):
(Inspector::InjectedScript::getCollectionEntries):
(Inspector::InjectedScript::saveResult):
(Inspector::InjectedScript::wrapCallFrames const):
(Inspector::InjectedScript::wrapObject const):
(Inspector::InjectedScript::wrapJSONString const):
(Inspector::InjectedScript::wrapTable const):
(Inspector::InjectedScript::previewValue const):
(Inspector::InjectedScript::setEventValue):
(Inspector::InjectedScript::clearEventValue):
(Inspector::InjectedScript::setExceptionValue):
(Inspector::InjectedScript::clearExceptionValue):
(Inspector::InjectedScript::findObjectById const):
(Inspector::InjectedScript::inspectObject):
(Inspector::InjectedScript::releaseObject):
(Inspector::InjectedScript::releaseObjectGroup):
(Inspector::InjectedScript::createCommandLineAPIObject const):
* Source/JavaScriptCore/inspector/InjectedScript.h:
* Source/JavaScriptCore/inspector/InjectedScriptBase.cpp:
(Inspector::jsToInspectorValue):
(Inspector::toInspectorValue):
(Inspector::InjectedScriptBase::InjectedScriptBase):
(Inspector::InjectedScriptBase::hasAccessToInspectedScriptState const):
(Inspector::InjectedScriptBase::injectedScriptObject const):
(Inspector::InjectedScriptBase::callFunctionWithEvalEnabled const):
(Inspector::InjectedScriptBase::makeCall):
(Inspector::InjectedScriptBase::makeEvalCall):
(Inspector::InjectedScriptBase::makeAsyncCall):
* Source/JavaScriptCore/inspector/InjectedScriptBase.h:
* Source/JavaScriptCore/inspector/InjectedScriptManager.cpp:
(Inspector::InjectedScriptManager::injectedScriptFor):
* Source/JavaScriptCore/inspector/InjectedScriptModule.cpp:
(Inspector::InjectedScriptModule::ensureInjected):
* Source/JavaScriptCore/inspector/ScriptFunctionCall.cpp: Renamed from Source/JavaScriptCore/bindings/ScriptFunctionCall.cpp.
(Inspector::ScriptCallArgumentHandler::appendArgument):
(Inspector::ScriptFunctionCall::ScriptFunctionCall):
(Inspector::ScriptFunctionCall::call):
* Source/JavaScriptCore/inspector/ScriptFunctionCall.h: Renamed from Source/JavaScriptCore/bindings/ScriptFunctionCall.h.
* Source/WebCore/html/HTMLMediaElement.cpp:
* Source/WebCore/inspector/CommandLineAPIHost.cpp:
* Source/WebCore/inspector/InspectorFrontendHost.cpp:
(WebCore::FrontendMenuProvider::create):
(WebCore::FrontendMenuProvider::disconnect):
(WebCore::FrontendMenuProvider::FrontendMenuProvider):
(WebCore::InspectorFrontendHost::showContextMenu):

Canonical link: <a href="https://commits.webkit.org/263625@main">https://commits.webkit.org/263625@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/34f3ac541c70c45018caf31da0a39d779c1132b6

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/5268 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/5404 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/5590 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/6799 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/5309 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/5265 "Passed tests") | [  ~~🛠 ios-sim~~](https://ews-build.webkit.org/#/builders/23/builds/5629 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/5382 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/5836 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/5364 "Passed tests") | [❌ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/5439 "layout-tests (failure)") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/4722 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/6821 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/2921 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/4725 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/11257 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/4359 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/4785 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/4801 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/6414 "Passed tests") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/4851 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/5221 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/4288 "Passed tests") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/5368 "Built successfully") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/4689 "Built successfully") | | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/1350 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/1262 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/8782 "Built successfully") | | [✅ 🛠 jsc-mips](https://ews-build.webkit.org/#/builders/24/builds/5520 "Built successfully") | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/5051 "Built successfully") | | [✅ 🧪 jsc-mips-tests](https://ews-build.webkit.org/#/builders/3/builds/1468 "Passed tests") | 
<!--EWS-Status-Bubble-End-->

darinadler commented on Source/JavaScriptCore/inspector/InjectedScriptBase.cpp:
```diff
@@ -41,14 +42,77 @@
+        return JSON::Value::null();
+    if (value.isBoolean())
+        return JSON::Value::create(value.asBoolean());
+    if (value.isNumber() && value.isDouble())
```
This is moved code, but I noticed that this should be just `value.isDouble()`.

Constellation replied:
Fixed.

darinadler commented on Source/JavaScriptCore/inspector/InjectedScriptBase.cpp:
```diff
@@ -41,14 +42,77 @@
+        return JSON::Value::create(value.asBoolean());
+    if (value.isNumber() && value.isDouble())
+        return JSON::Value::create(value.asNumber());
+    if (value.isNumber() && value.isAnyInt())
```
And this should be `value.isInt32()`.

Constellation replied:
Changed.

darinadler commented on Source/JavaScriptCore/inspector/InjectedScriptBase.cpp:
```diff
@@ -41,14 +42,77 @@
+    if (value.isBoolean())
+        return JSON::Value::create(value.asBoolean());
+    if (value.isNumber() && value.isDouble())
+        return JSON::Value::create(value.asNumber());
```
And this should be `value.asDouble()`.

Constellation replied:
Changed.

darinadler commented on Source/JavaScriptCore/inspector/InjectedScriptBase.cpp:
```diff
@@ -41,14 +42,77 @@
+    if (value.isNumber() && value.isDouble())
+        return JSON::Value::create(value.asNumber());
+    if (value.isNumber() && value.isAnyInt())
+        return JSON::Value::create(static_cast<int>(value.asAnyInt()));
```
And this should be `value.asInt32()` without a `static_cast`.

Constellation replied:
Changed.

darinadler commented on Source/JavaScriptCore/inspector/InjectedScriptBase.h:
```diff
@@ -49,34 +49,37 @@ namespace Inspector {
 
     String m_name;
-    Deprecated::ScriptObject m_injectedScriptObject;
+    JSC::JSGlobalObject* m_globalObject { nullptr };
```
Is it OK to have this as a raw pointer? Do we need to protect from garbage collection?

Constellation replied:
Yes, it is actually OK since this is held by JSGlobalObject

Constellation committed:
Remove JavaScriptCore/bindings
https://bugs.webkit.org/show_bug.cgi?id=256033
rdar://problem/108596608

Reviewed by Darin Adler.

All files under JavaScriptCore/bindings were Deprecated::.

1. Remove ScriptValue and ScriptObject. Use Strong<> directly. We do not need these wrappers anymore.
2. Move ScriptFunctionCall to inspector/ since it is only used for inspector code, and still it is useful.

* Source/JavaScriptCore/CMakeLists.txt:
* Source/JavaScriptCore/JavaScriptCore.xcodeproj/project.pbxproj:
* Source/JavaScriptCore/Sources.txt:
* Source/JavaScriptCore/bindings/ScriptObject.cpp: Removed.
* Source/JavaScriptCore/bindings/ScriptObject.h: Removed.
* Source/JavaScriptCore/bindings/ScriptValue.cpp: Removed.
* Source/JavaScriptCore/bindings/ScriptValue.h: Removed.
* Source/JavaScriptCore/inspector/InjectedScript.cpp:
(Inspector::InjectedScript::InjectedScript):
(Inspector::InjectedScript::execute):
(Inspector::InjectedScript::evaluate):
(Inspector::InjectedScript::awaitPromise):
(Inspector::InjectedScript::callFunctionOn):
(Inspector::InjectedScript::evaluateOnCallFrame):
(Inspector::InjectedScript::getFunctionDetails):
(Inspector::InjectedScript::functionDetails):
(Inspector::InjectedScript::getPreview):
(Inspector::InjectedScript::getProperties):
(Inspector::InjectedScript::getDisplayableProperties):
(Inspector::InjectedScript::getInternalProperties):
(Inspector::InjectedScript::getCollectionEntries):
(Inspector::InjectedScript::saveResult):
(Inspector::InjectedScript::wrapCallFrames const):
(Inspector::InjectedScript::wrapObject const):
(Inspector::InjectedScript::wrapJSONString const):
(Inspector::InjectedScript::wrapTable const):
(Inspector::InjectedScript::previewValue const):
(Inspector::InjectedScript::setEventValue):
(Inspector::InjectedScript::clearEventValue):
(Inspector::InjectedScript::setExceptionValue):
(Inspector::InjectedScript::clearExceptionValue):
(Inspector::InjectedScript::findObjectById const):
(Inspector::InjectedScript::inspectObject):
(Inspector::InjectedScript::releaseObject):
(Inspector::InjectedScript::releaseObjectGroup):
(Inspector::InjectedScript::createCommandLineAPIObject const):
* Source/JavaScriptCore/inspector/InjectedScript.h:
* Source/JavaScriptCore/inspector/InjectedScriptBase.cpp:
(Inspector::jsToInspectorValue):
(Inspector::toInspectorValue):
(Inspector::InjectedScriptBase::InjectedScriptBase):
(Inspector::InjectedScriptBase::hasAccessToInspectedScriptState const):
(Inspector::InjectedScriptBase::injectedScriptObject const):
(Inspector::InjectedScriptBase::callFunctionWithEvalEnabled const):
(Inspector::InjectedScriptBase::makeCall):
(Inspector::InjectedScriptBase::makeEvalCall):
(Inspector::InjectedScriptBase::makeAsyncCall):
* Source/JavaScriptCore/inspector/InjectedScriptBase.h:
* Source/JavaScriptCore/inspector/InjectedScriptManager.cpp:
(Inspector::InjectedScriptManager::injectedScriptFor):
* Source/JavaScriptCore/inspector/InjectedScriptModule.cpp:
(Inspector::InjectedScriptModule::ensureInjected):
* Source/JavaScriptCore/inspector/ScriptFunctionCall.cpp: Renamed from Source/JavaScriptCore/bindings/ScriptFunctionCall.cpp.
(Inspector::ScriptCallArgumentHandler::appendArgument):
(Inspector::ScriptFunctionCall::ScriptFunctionCall):
(Inspector::ScriptFunctionCall::call):
* Source/JavaScriptCore/inspector/ScriptFunctionCall.h: Renamed from Source/JavaScriptCore/bindings/ScriptFunctionCall.h.
* Source/WebCore/html/HTMLMediaElement.cpp:
* Source/WebCore/inspector/CommandLineAPIHost.cpp:
* Source/WebCore/inspector/InspectorFrontendHost.cpp:
(WebCore::FrontendMenuProvider::create):
(WebCore::FrontendMenuProvider::disconnect):
(WebCore::FrontendMenuProvider::FrontendMenuProvider):
(WebCore::InspectorFrontendHost::showContextMenu):

Canonical link: https://commits.webkit.org/263625@main
